l10n_br_account/models/account_invoice_line.py

```diff
@@ -519,3 +388,42 @@ def _onchange_mark_recompute_taxes(self):
         user must be able to set a custom value.
         """
         return super()._onchange_mark_recompute_taxes()
+
+    # In localization, until v12.0, the accounting entry for revenue was always done
+    # with the value of all taxes included. The method was rewritten to take into
+    # account the price_total instead of the sub_total.
+    @api.model
+    def _get_fields_onchange_subtotal_model(
+        self, price_subtotal, move_type, currency, company, date
+    ):
+        """This method is used to recompute the values of 'amount_currency', 'debit',
+        'credit' due to a change made in some business fields (affecting the
+        'price_subtotal' field).
+
+        :param price_subtotal:  The untaxed amount.
+        :param move_type:       The type of the move.
+        :param currency:        The line's currency.
+        :param company:         The move's company.
+        :param date:            The move's date.
+        :return:                A dictionary containing 'debit', 'credit', 'amount_currency'.
+        """
+        if move_type in self.move_id.get_outbound_types():
+            sign = 1
+        elif move_type in self.move_id.get_inbound_types():
+            sign = -1
+        else:
+            sign = 1
+
+        amount_currency = self.price_total * sign
```

Comment:
obrigado por ter se ligado ao problema @felipemotter e @antoniospneto . Porem eu acho que esse codigo nem sempre vai funcionar. No create em especial veja como o método é chamado https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L4189 ou seja o self vai ser um recordset vazio e self.price_total não deveria funcionar. Tem que testar bem, mas eu alerto com isso porque foi esse mesmo codigo que eu tive que mexer no caso dos movimentos sem financeiro e eu tive esse problema com o self vazio no create. Veja que no caso eles pegam o valor do vals e não do self: `vals['price_subtotal']`.

Outro pronto: seria interessante testar se o move tem um document_type_id e se não tiver ou chamar o super ou talvez ter o código original num if.

Reply:
Obrigado por levantar esse ponto @rvalyi. Ficamos com medo disso também, porém estamos com medo de que a alteração de códigos tenha que ser extremamente profunda, alterando todos os métodos que acabam chamando este, para enviar o price_total ao invés do subtotal. Você vê uma outra solução além desta?

Reply:
misericórdia, aqui ta 54°C de sensação thermica... dificil mergulhar nisso agora... Mas eh, deixa ver o que tava testando para as remessas que eu imagino que vai ter que ser o mesmo tipo de solução. Daqui pouco eu retorno, talvez amanha 
 ou sexta porem.

Reply:
@rvalyi como na v16 essa parte vai mudar tudo, uma forma mais fácil de fazer isso seria fazer um override no create e no write, e depois da execução do super, a gente força a chamada desse método de novo. 

Bonito e eficiente não é, mas seria uma medida que seria revista na 16, já que houve um refactor em tudo isso na 16.